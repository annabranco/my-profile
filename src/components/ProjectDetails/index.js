import React from 'react';
import { useSelector } from 'react-redux';
import { bool } from 'prop-types';
import { currentLanguageSelector } from '../../redux/selectors';
import { THUMBNAILS } from '../../assets/images';
import { projectsPropType } from '../../types';
import {
  Description,
  Link,
  ProjectItem,
  RepoIcon,
  SeeIcon,
  Thumbnail,
  Title
} from './styles';

const a = THUMBNAILS;

const ProjectDetails = ({ displayThumbnails, project }) => {
  const languageSelected = useSelector(currentLanguageSelector);

  return (
    <ProjectItem key={project.title}>
      <Link href={project.url} rel="noopener noreferrer" target="_Blank">
        <Title>
          {project.title}
          <SeeIcon className="far fa-eye" />
        </Title>
      </Link>
      <Description>{project.description[languageSelected]}</Description>

      <Link href={project.repo} rel="noopener noreferrer" target="_Blank">
        <p className="project__repo">
          <RepoIcon className="fab fa-github-alt" />
          {project.repo.replace('https://github.com', '')}
        </p>
      </Link>
      {displayThumbnails && (
        <Link href={project.url} rel="noopener noreferrer" target="_Blank">
          <Thumbnail src={THUMBNAILS[project.thumbnail]} alt={project.title} />
        </Link>
      )}
    </ProjectItem>
  );
};

ProjectDetails.propTypes = {
  displayThumbnails: bool.isRequired,
  project: projectsPropType.isRequired
};

export default ProjectDetails;
