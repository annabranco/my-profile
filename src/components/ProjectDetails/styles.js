import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorBlueDark,
  colorBlack,
  fontTitle,
  colorGrayDark
} from '../../styles/theme';

export const Description = styled.p`
  margin-bottom: 20px;
  line-height: 1.4;
  text-align: justify;
`;
Description.displayName = '--- Description ---';

export const Link = styled.a`
  text-decoration: none;
  &:hover {
    text-decoration: none;
  }
`;
Link.displayName = '--- Link ---';

export const ProjectItem = styled.div`
  height: auto;
`;
ProjectItem.displayName = '--- Project Item ---';

export const RepoIcon = styled.i`
  opacity: 0.4;
  margin-right: 4px;
  vertical-align: middle;
  font-size: 1.3rem;
  color: ${colorBlack};
`;
RepoIcon.displayName = '--- Repo Icon ---';

export const SeeIcon = styled.i`
  margin-left: 10px;
  opacity: 0.4;
  color: ${colorBlack};
`;
SeeIcon.displayName = '--- SeeIcon ---';

export const Thumbnail = styled.img`
  border: 1px solid ${colorBlack};
  border-radius: 5px;
  margin-top: 10px;
  width: 100%;
  box-shadow: 0 3px 7px 0 ${rgba(colorGrayDark, 0.8)};
`;
Thumbnail.displayName = '--- Thumbnail ---';

export const Title = styled.h3`
  margin: 10px 0;
  font-family: ${fontTitle};
  font-size: 1.3rem;
  font-weight: bold;
  color: ${colorBlueDark};
`;
Title.displayName = '--- Title ---';
