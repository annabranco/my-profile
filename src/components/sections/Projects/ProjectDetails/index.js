import React from 'react';
import { useSelector } from 'react-redux';
import { number } from 'prop-types';
import { currentLanguageSelector } from '../../../../redux/selectors';
import { isDesktop } from '../../../../utils/device';
import { THUMBNAILS } from '../../../../assets/images';
import { projectsPropType } from '../../../../types';
import {
  Description,
  Link,
  ProjectItem,
  RepoIcon,
  RepoUrl,
  SeeIcon,
  Thumbnail,
  Title
} from './styles';

const ProjectDetails = ({ actualPage, project }) => {
  const languageSelected = useSelector(currentLanguageSelector);

  return (
    <ProjectItem actualPage={actualPage} key={project.title}>
      {isDesktop && (
        <>
          <Link href={project.url} rel="noopener noreferrer" target="_Blank">
            <Title>
              {project.title}
              <SeeIcon className="far fa-eye" />
            </Title>
          </Link>
          <Description>{project.description[languageSelected]}</Description>
          <Link href={project.repo} rel="noopener noreferrer" target="_Blank">
            <RepoUrl className="project__repo">
              <RepoIcon className="fab fa-github-alt" />
              {project.repo.replace('https://github.com', '')}
            </RepoUrl>
          </Link>
        </>
      )}
      <Link
        href={isDesktop ? project.url : project.repo}
        rel="noopener noreferrer"
        target="_Blank"
      >
        <Thumbnail src={THUMBNAILS[project.thumbnail]} alt={project.title} />
        {!isDesktop && <Title>{project.title}</Title>}
      </Link>
    </ProjectItem>
  );
};

ProjectDetails.propTypes = {
  actualPage: number.isRequired,
  project: projectsPropType.isRequired
};

export default ProjectDetails;
