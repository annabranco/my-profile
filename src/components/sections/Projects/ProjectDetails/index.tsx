import { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { number } from 'prop-types';
import { IProjects } from '../../../../types/interfaces';
import { currentLanguageSelector } from '../../../../redux/selectors';
import { isDesktop } from '../../../../utils/device';
import { THUMBNAILS } from '../../../../assets/images';
import { projectsPropType } from '../../../../types/propTypes';
import {
  Description,
  Link,
  ProjectItem,
  RepoIcon,
  RepoUrl,
  SeeIcon,
  Thumbnail,
  Title
} from './styles';

interface Props {
  actualPage: number;
  project: IProjects;
}

const ProjectDetails = ({ actualPage, project }: Props): ReactElement => {
  const languageSelected = useSelector(currentLanguageSelector);

  return (
    <ProjectItem actualPage={actualPage} key={project.title}>
      {isDesktop && (
        <>
          <Link
            href={project.url}
            data-e2e-id={`projects-title_${project.title}`}
            rel="noopener noreferrer"
            target="_Blank"
          >
            <Title>
              {project.title}
              <SeeIcon className="far fa-eye" />
            </Title>
          </Link>
          <Description>{project.description[languageSelected]}</Description>
          <Link
            href={project.repo}
            data-e2e-id={`projects-repo_${project.title}`}
            rel="noopener noreferrer"
            target="_Blank"
          >
            <RepoUrl className="project__repo">
              <RepoIcon className="fab fa-github-alt" />
              {project.repo.replace('https://github.com', '')}
            </RepoUrl>
          </Link>
        </>
      )}
      <Link
        data-e2e-id={`projects-thumb_${project.title}`}
        href={isDesktop ? project.url : project.repo}
        rel="noopener noreferrer"
        target="_Blank"
      >
        <Thumbnail
          src={THUMBNAILS[project.thumbnail as keyof typeof THUMBNAILS]}
          alt={project.title}
        />
        {!isDesktop && <Title>{project.title}</Title>}
      </Link>
    </ProjectItem>
  );
};

ProjectDetails.propTypes = {
  actualPage: number.isRequired,
  project: projectsPropType.isRequired
};

export default ProjectDetails;
