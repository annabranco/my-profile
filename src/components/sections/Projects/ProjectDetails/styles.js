import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorBlueDark,
  colorBlack,
  fontTitle,
  colorGrayDark,
  colorWhite
} from '../../../../styles/theme';

export const Description = styled.p`
  margin-bottom: 10px;
  line-height: 1.4;
  text-align: justify;
  font-size: 0.8rem;

  @media all and (min-width: 768px) {
    margin-bottom: 20px;
    font-size: 1rem;
  }
`;
Description.displayName = 'Description';

export const Link = styled.a`
  text-decoration: none;
  &:hover {
    text-decoration: none;
  }
`;
Link.displayName = 'Link';

export const ProjectItem = styled.div`
  height: auto;
`;
ProjectItem.displayName = 'Project Item';

export const RepoIcon = styled.i`
  opacity: 0.4;
  margin-right: 4px;
  vertical-align: middle;
  font-size: 1.3rem;
  color: ${colorBlack};
`;
RepoIcon.displayName = 'Repo Icon';

export const RepoUrl = styled.p`
  margin: 0;
  text-align: right;

  @media all and (min-width: 768px) {
    text-align: left;
  }
`;
RepoIcon.displayName = 'Repo Icon';

export const SeeIcon = styled.i`
  margin-left: 10px;
  opacity: 0.4;
  color: ${colorBlack};
  -webkit-text-stroke: 1px black;

  &:hover {
    color: orange;
  }
`;
SeeIcon.displayName = 'SeeIcon';

export const Thumbnail = styled.img`
  z-index: 5;
  border: 1px solid ${colorBlack};
  border-radius: 5px;
  margin-top: 10px;
  margin-left: 20px;
  width: 90%;
  box-shadow: 0 3px 7px 0 ${rgba(colorGrayDark, 0.8)};
  transition: all 1s ease;

  &:hover {
    box-shadow: 0 0 10px 2px rgba(248, 148, 6, 0.3);
    filter: drop-shadow(2px 4px 6px black) brightness(1.2);
    transform: scale(1.05);
  }

  @media all and (min-width: 768px) {
    margin-left: 0;
  }
`;
Thumbnail.displayName = 'Thumbnail';

export const Title = styled.h3`
  z-index: 50;
  position: relative;
  margin: -10px 0 10px;
  font-family: ${fontTitle};
  font-size: 1.3rem;
  font-weight: bold;
  color: ${colorWhite};
  background: ${colorBlueDark};
  border-radius: 8px 3px 3px 8px;
  padding-left: 20px;
  box-shadow: 1px 2px 10px 0 gray;

  @media all and (min-width: 768px) {
    position: unset;
    margin: 10px 0;
    background: none;
    box-shadow: none;
    padding-left: 0;
    color: ${colorBlueDark};
  } ;
`;
Title.displayName = 'Title';
