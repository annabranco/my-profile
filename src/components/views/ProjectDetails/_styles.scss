.project__item {
  height : auto;
}

.project__title {
  margin      : 10px 0;
  font-family : $font__title;
  font-size   : 1.3rem;
  font-weight : bold;
  color       : $blue--dark;
}

.project__title-icon {
  opacity : 0.4;
  color   : $black;
}

.project__thumbnail {
  border        : 1px solid $black;
  border-radius : 5px;
  margin-top    : 10px;
  width         : 100%;
  box-shadow    : 0 3px 7px 0 rgba($gray-dark, 0.8);
}

.project__description {
  margin-bottom : 20px;
  line-height   : 1.4;
  text-align    : justify;
}

.project__url-a {
  text-decoration : none;

  &:hover {
    text-decoration : none;
  }
}

.project__repo--icon {
  opacity        : 0.4;
  margin-right   : 4px;
  vertical-align : middle;
  font-size      : 1.3rem;
  color          : $black;
}
