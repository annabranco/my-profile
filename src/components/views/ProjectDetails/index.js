import React from 'react';
import { bool, string } from 'prop-types';
import {
  ProjectItem,
  Link,
  Title,
  SeeIcon,
  Description,
  RepoIcon,
  Thumbnail
} from './styles';
import { projectsPropType } from '../../../types';

const ProjectDetails = ({ displayThumbnails, language, project }) => (
  <ProjectItem key={project.title}>
    <Link href={project.url} rel="noopener noreferrer" target="_Blank">
      <Title>
        {project.title}
        <SeeIcon className="far fa-eye" />
      </Title>
    </Link>
    <Description>{project.description[language]}</Description>

    <Link href={project.repo} rel="noopener noreferrer" target="_Blank">
      <p className="project__repo">
        <RepoIcon className="fab fa-github-alt" />
        {project.repo.replace('https://github.com', '')}
      </p>
    </Link>
    {displayThumbnails && (
      <Link href={project.url} rel="noopener noreferrer" target="_Blank">
        <Thumbnail src={project.thumbnail} alt={project.title} />
      </Link>
    )}
  </ProjectItem>
);

ProjectDetails.propTypes = {
  displayThumbnails: bool.isRequired,
  language: string.isRequired,
  project: projectsPropType.isRequired
};

export default ProjectDetails;
