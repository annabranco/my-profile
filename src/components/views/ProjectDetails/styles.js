import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorBlueDark,
  colorBlack,
  fontTitle,
  colorGrayDark
} from '../../../styles/theme';

export const ProjectItem = styled.div`
  height: auto;
`;

export const ProjectLink = styled.a`
  text-decoration: none;
  &:hover {
    text-decoration: none;
  }
`;

export const ProjectTitle = styled.h3`
  margin: 10px 0;
  font-family: ${fontTitle};
  font-size: 1.3rem;
  font-weight: bold;
  color: ${colorBlueDark};
`;

export const ProjectTitleIcon = styled.i`
  opacity: 0.4;
  color: ${colorBlack};
`;

export const ProjectDescription = styled.p`
  margin-bottom: 20px;
  line-height: 1.4;
  text-align: justify;
`;

export const ProjectRepoIcon = styled.i`
  opacity: 0.4;
  margin-right: 4px;
  vertical-align: middle;
  font-size: 1.3rem;
  color: ${colorBlack};
`;

export const ProjectThumbnail = styled.img`
  border: 1px solid ${colorBlack};
  border-radius: 5px;
  margin-top: 10px;
  width: 100%;
  box-shadow: 0 3px 7px 0 ${rgba(colorGrayDark, 0.8)};
`;
