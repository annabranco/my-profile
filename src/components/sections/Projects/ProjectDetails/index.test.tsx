import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import ProjectDetails from '.';
import { getMockState, setupMockProvider } from '../../../../tests/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< ProjectDetails >', () => {
  const mockProject = getMockState({
    language: 'en'
  }).projects[0];
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <ProjectDetails actualPage={1} project={mockProject} />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should render info about the project received by props, displaying  the name, thumbnail and link', () => {});

  xit('should render the project description on desktops', () => {});

  xit('should display an animation when displaying each pair of projects', () => {});

  xit('should translate all texts when the language is changed', () => {});
});
