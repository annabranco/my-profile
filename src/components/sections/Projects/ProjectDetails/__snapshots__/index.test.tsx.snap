// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ProjectDetails > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProjectDetails
    actualPage={1}
    project={
      Object {
        "description": Object {
          "en": "An  unofficial app to enhance players' experience while playing Zombicide tabletop game.",
          "es": "Una app no oficial para mejorar la experiencia del juego de mesa Zombicide.",
          "pt": "Una app nãp oficial para melhorar a experiência do jogo de tabuleiro Zombicide.",
        },
        "order": 1,
        "repo": "https://github.com/annabranco/zombicide-party",
        "thumbnail": "zombicide-party",
        "title": "Zombicide Party",
        "url": "https://annabranco.github.io/zombicide-party/",
      }
    }
  />
</Provider>
`;
